### ISSUE TITLE
Ability to disable notifications in authProvider.checkAuth function

### ISSUE THREAD

**Issue by panfiva** [marmelab/react-admin#5178]
**Is your feature request related to a problem? Please describe.**
When users are not authenticated, when accessing the application, they are redirected to the login page since `checkAuth` function fails. When `checkAuth` fails, react-admin also issues notification `Not authenticated` that is displayed in the login page.

Since we use Okta authentication, the application logs-in our users automatically, so this notification is not needed. Moreover, the entire logon process take about 1 seconds, after that users are redirected to the home page. The home page also displays `Not authenticated` warning, since the notification has not yet timed-out. This is very confusing for our users.

**Describe the solution you'd like**
We would like to be able to disable react-admin's notification that is triggered when `checkAuth` returns Promise.reject.

We would also like the ability to be able to issue custom notifications from authProvider functions and also issue messages, so if you could pass `dispatch` function and `notify` functions as properties, that will make issuing custom notifications easier.

**Describe alternatives you've considered**
The alternative solution that we have is to add the following code block to our login page to cancel the notification; however it would be better for everyone to have a built-in ability to stop the notification message.
```js
import { hideNotification } from 'react-admin'
import { useDispatch} from 'react-redux'

export const LoginPage = function (props) {
...
   const dispatch = useDispatch()
   useEffect(() => {
      dispatch(hideNotification())
     }, [])
...
}
```

**Comment by djhi** [comment]
Hi, thanks for reporting.

We would gladly accept a pull request for the `useCheckAuth` and `useLogoutIfAccessDenied` which already check whether the `authProvider` returned a `redirectTo`. They could also check for `disableNotification` value and act accordingly.

**Comment by fzaninotto** [comment]
Fixed by #5255
